### PR TITLE
Non HW unit test CI

### PR DESCRIPTION
**Pull request by tagoylo**
Updated unit tests for downloader and cli pieces that does not need hardware.
Added github workflow that runs the test.

### CODE REVIEW

File: tests/nebula_config/nebula.yaml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    - hdl_folder: fmcomms2_zc702
+  driver-config:
+    - iio_device_names:
+      - ad7291
+      - ad9361-phy
+      - adf4351-udc-tx-pmod
+      - adf4351-udc-rx-pmod
+      - cf-ad9361-dds-core-lpc
+      - cf-ad9361-lpc
+  netbox-config:
```
Comment (kimpaller) [inline tests/nebula_config/nebula.yaml:19]: is it ok to exclude netbox-config section from the test nebula config file? It exposes network information that might compromise security. Or just replace values with dummies.

Comment (kimpaller) [reply tests/nebula_config/nebula.yaml:19]: if its possible mask every IP information or define its value at runtime using  environment variables

File: tests/test_cli.py [hunk 2]
```diff
@@ -6,27 +7,92 @@
 from fabric import Connection as con
 from nebula import pdu, uart
 
+# must include -s
+
+
+@pytest.fixture(autouse=True)
```
Comment (kimpaller) [inline tests/test_cli.py:13]: might not necessary to use autouse=True at this instance.

Comment (tagoylo) [reply tests/test_cli.py:13]: This is necessary to ensure that downloaded files from previous tests are deleted.

File: tests/nebula_config/nebula.yaml (hunk 1 above)
Comment (tagoylo) [reply tests/nebula_config/nebula.yaml]: [5b64bb9](https://github.com/sdgtt/nebula/pull/74/commits/5b64bb92b91374036f70a946c328f3197489fcf1) Removed network configs and other configs that are not needed through this commit.

**Pull request merged**